morphia/src/main/java/org/mongodb/morphia/IndexBuilder.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright 2016 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mongodb.morphia;
+
+import org.mongodb.morphia.annotations.Field;
+import org.mongodb.morphia.annotations.Index;
+import org.mongodb.morphia.annotations.IndexOptions;
+import org.mongodb.morphia.utils.IndexType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@SuppressWarnings("deprecation")
+class IndexBuilder extends AnnotationBuilder<Index> implements Index {
+    IndexBuilder() {
+    }
+
+    IndexBuilder(final Index original) {
+        super(original);
+    }
+
+    static Index normalize(final Index index) {
+        return index.fields().length != 0
+               ? index
+               : new IndexBuilder()
+                   .migrate(index);
+    }
+
+    @Override
+    public Class<Index> annotationType() {
+        return Index.class;
+    }
+
+    public IndexBuilder fields(final String fields) {
+        return fields(parseFieldsString(fields));
+    }
+
+    @Override
+    public Field[] fields() {
+        return get("fields");
+    }
+
+    @Override
+    public IndexOptions options() {
+        return get("options");
+    }
+
+    @Override
+    public boolean background() {
+        return get("background");
+    }
+
+    @Override
+    public boolean disableValidation() {
+        return get("disableValidation");
+    }
+
+    @Override
+    public boolean dropDups() {
+        return get("dropDups");
+    }
+
+    @Override
+    public int expireAfterSeconds() {
+        return get("expireAfterSeconds");
+    }
+
+    @Override
+    public String name() {
+        return get("name");
+    }
+
+    @Override
+    public boolean sparse() {
+        return get("sparse");
+    }
+
+    @Override
+    public boolean unique() {
+        return get("unique");
+    }
+
+    @Override
+    public String value() {
+        return get("value");
+    }
+
+    private List<Field> parseFieldsString(final String str) {
```

Comment:
Thats a bit of a scary method.  Is the string something the user will define? Does there need to be any error checking / validation?